certbot/docs/install.rst

```diff
@@ -303,6 +235,33 @@ OS packaging is an ongoing effort. If you'd like to package
 Certbot for your distribution of choice please have a
 look at the :doc:`packaging`.
 
+.. _certbot-auto:
+
+Certbot-Auto
+------------
+
+We used to have a shell script named ``certbot-auto`` to help people install
+Certbot on UNIX operating systems, however, this script is no longer supported.
+
+Problems with Python virtual environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When using ``certbot-auto`` on a low memory system such as VPS with less than
+512MB of RAM, the required dependencies of Certbot may fail to build.  This can
```

Comment:
What is the reason to keep this section around? 

Are there other (non-development) circumstances in which it is relevant, like the `pipx` proposal perhaps?

If so, shall we reword it to not mention `certbot-auto`?

Reply:
My motivation for keeping this around was indeed to not break the anchor, especially when `certbot-auto` isn't quite fully deprecated yet and it links to it. See https://github.com/certbot/certbot/blob/5151e2afee98adebce061215ecda6a739c4f2ecd/certbot-auto#L1730

It currently does not affect any of our non-development installation instructions. This problem can occur when compiling cryptography/cffi and in snaps, Docker, and OS packages it is precompiled. (I suppose a user of a source based distribution like Gentoo could hit this, although not in a Python virtual environment as described in the title of this section, but I've never seen this in practice.)

This could affect whatever becomes of our `pipx` proposal, but I think this is largely a resolved problem. This used to be a very common issue (see https://github.com/certbot/certbot/issues/1081) but since then `cryptography` and `cffi` have shipped precompiled wheels for amd64 and aarch64. We could maybe keep this around in anticipation of the `pipx` instructions in case anyone hits this issue, but I personally think this is rare enough that it adds more noise than it's worth. I'm interested in your opinion, but mine is we should make this certbot-auto specific and delete it at some point, but if we see (m)any users confused by this in the future, we could always add the section back or generalize it.

Reply:
Ah, OK.

I was under the impression that there were no longer any systems that would call `pip install` anymore, unless they purposefully evaded the deprecation by cloning from an earlier tag or something.

If the code still runs on some end-user systems, I think it's fine to keep it as-is.

Otherwise, I think that fully removing it is appropriate, as it has no end-user audience.

Reply:
certbot-auto will be fully deprecated in our next release, but since we haven't released that change yet, I have a slight preference to keep this around so we don't have to worry about accidentally breaking the anchor on the website before our next release.